File: AdvancePython/Day-3 Advanced OOP concepts/Task-4.py
# Task 4:
class Employee:
    class Address:
        def __init__(self, country: str) -> None:
            self.country: str = country

        def display(self) -> None:
            print(f"Address: {self.country}")

    def __init__(self, name: str, country: str = "India") -> None:
        self.name = name
        self.addr = self.Address(country)

    def show(self) -> None:
        print(f"Name: {self.name}")
        self.addr.display()


e1 = Employee("Virat")
e1.show()

File: node.js/Day-3/Task-4.js
// Task 4:

class Employee {
    constructor() {
        this.name = "Virat";
        this.add = new Address();
    }

    info() {
        console.log(`Name: ${this.name}`);
        this.add.display();
    }
}

class Address {
    constructor() {
        this.country = "India";
    }
    
    display() {
        console.log(`Address: ${this.country}`);
    }
}

const emp = new Employee();
emp.info();